modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/base_resource_impl.ftl

```diff
@@ -508,31 +511,141 @@ public abstract class Base${schemaName}ResourceImpl
 
 			<#if createStrategies?seq_contains("UPSERT")>
 				if ("UPSERT".equalsIgnoreCase(createStrategy)) {
-					${schemaVarName}UnsafeConsumer = ${schemaVarName} -> ${putByERCBatchJavaMethodSignature.methodName}(
-
-					<#list putByERCBatchJavaMethodSignature.javaMethodParameters as javaMethodParameter>
-						<#if stringUtil.equals(javaMethodParameter.parameterName, "externalReferenceCode")>
-							${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}()
-						<#elseif putByERCBatchJavaMethodSignature.parentSchemaName?? && stringUtil.equals(javaMethodParameter.parameterName, putByERCBatchJavaMethodSignature.parentSchemaName!?uncap_first + "Id")>
-							<#if properties?keys?seq_contains(javaMethodParameter.parameterName)>
-								${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}() != null ?
-								${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}() :
+					String updateStrategy = (String) parameters.getOrDefault("updateStrategy", "UPDATE");
+
+				<#if updateStrategies?seq_contains("UPDATE") && putByERCBatchJavaMethodSignature??>
+						if("UPDATE".equals(updateStrategy)) {
+							${schemaVarName}UnsafeConsumer = ${schemaVarName} -> ${putByERCBatchJavaMethodSignature.methodName}(
+
+							<#list putByERCBatchJavaMethodSignature.javaMethodParameters as javaMethodParameter>
+								<#if stringUtil.equals(javaMethodParameter.parameterName, "externalReferenceCode")>
+									${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}()
+								<#elseif putByERCBatchJavaMethodSignature.parentSchemaName?? && stringUtil.equals(javaMethodParameter.parameterName, putByERCBatchJavaMethodSignature.parentSchemaName!?uncap_first + "Id")>
+									<#if properties?keys?seq_contains(javaMethodParameter.parameterName)>
+										${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}() != null ?
+										${schemaVarName}.get${javaMethodParameter.parameterName?cap_first}() :
+									</#if>
+
+									<@castParameters
+										type=javaMethodParameter.parameterType
+										value="${javaMethodParameter.parameterName}"
+									/>
+								<#elseif stringUtil.equals(javaMethodParameter.parameterName, schemaVarName)>
+									${schemaVarName}
+								<#else>
+									null
+								</#if>
+
+								<#sep>, </#sep>
+							</#list>
+
+							);
+						}
+					</#if>
```

Comment:
The tabs are not properly formatted. Same in the following if clause

Reply:
Fixed!